lib/commands/device/move.ts

```diff
@@ -107,14 +113,18 @@ export default class DeviceMoveCmd extends Command {
 		}
 
 		// Get destination application
-		const application =
+		const applicationIdentifier =
 			options.application ||
 			(await this.interactivelySelectApplication(balena, devices));
```

Comment:
Probably for another PR... I've noticed that `interactivelySelectApplication()` returns the app name instead of the app slug or the application object -- i.e. the CLI code is doing what this PR advises users not to do: selecting apps by name rather than slug. I think it's possible that users would select the right app via the dropdown menus, only for the CLI to then error with "Application is ambiguous". 🤦  The best would probably be for the application object to be returned, then the caller decides how to use it.

Reply:
I'll look at this in this PR.